src/SOS/SOS.Extensions/TargetFromFromDebuggerServices.cs

```diff
@@ -94,7 +95,43 @@ internal TargetFromDebuggerServices(DebuggerServices debuggerServices, IHost hos
                 return memoryService;
             });
 
+            // Add optional crash info service (currently only for Native AOT).
+            _serviceContainerFactory.AddServiceFactory<ICrashInfoService>((services) => CreateCrashInfoService(services, debuggerServices));
+            OnFlushEvent.Register(() => FlushService<ICrashInfoService>());
+
             Finished();
         }
+
+        private unsafe ICrashInfoService CreateCrashInfoService(IServiceProvider services, DebuggerServices debuggerServices)
+        {
+            // For Linux/OSX dumps loaded under dbgeng the GetLastException API doesn't return the necessary information
+            if (Host.HostType == HostType.DbgEng && (OperatingSystem == OSPlatform.Linux || OperatingSystem == OSPlatform.OSX))
+            {
+                return SpecialDiagInfo.CreateCrashInfoService(services);
+            }
+            HResult hr = debuggerServices.GetLastException(out uint processId, out int threadIndex, out EXCEPTION_RECORD64 exceptionRecord);
+            if (hr.IsOK)
+            {
+                if (exceptionRecord.ExceptionCode == CrashInfoService.STATUS_STACK_BUFFER_OVERRUN &&
+                    exceptionRecord.NumberParameters >= 4 &&
+                    exceptionRecord.ExceptionInformation[0] == CrashInfoService.FAST_FAIL_EXCEPTION_DOTNET_AOT)
+                {
+                    uint hresult = (uint)exceptionRecord.ExceptionInformation[1];
+                    ulong triageBufferAddress = exceptionRecord.ExceptionInformation[2];
+                    int triageBufferSize = (int)exceptionRecord.ExceptionInformation[3];
+
+                    Span<byte> buffer = stackalloc byte[triageBufferSize];
```

Comment:
Similar to above, I wouldn't `stackalloc` based on anything coming from the exception record.